src/sagas/bluetoothSaga.js

```diff
@@ -0,0 +1,196 @@
+import {
+  take,
+  call,
+  put,
+  select,
+  all,
+  delay,
+  fork,
+  race,
+} from 'redux-saga/effects';
+import {
+  BLUETOOTH_BROADCASTER_TASK_STOPPED,
+  BLUETOOTH_SCANNER_TASK_STOPPED,
+  BLUETOOTH_START_UP_SUCCESS,
+  BLUETOOTH_STATE_CHANGED,
+  BLUETOOTH_TEAR_DOWN,
+  BLUETOOTH_TEAR_DOWN_SUCCESS,
+  CHECK_DEVICE_BLUETOOTH_STATE,
+  DEVICE_SCANNED,
+} from '../actions/bluetoothActions';
+import {NETWORK_STATE, START_DISCONNECTION} from '../actions/lasnActions';
+import {createBluetoothChannel} from '../channels/bluetoothChannel';
+import {createBluetoothScannerChannel} from '../channels/bluetoothScannerChannel';
+import {
+  BLUETOOTH_EVENT_TYPES,
+  BLUETOOTH_SCANNER_EVENT_TYPES,
+} from '../channels/events/bluetoothChannelEvents';
+import {selectDeviceId} from '../selectors/userSelectors';
+import {State as BLUETOOTH_STATE} from 'react-native-ble-plx';
+import {
+  broadcastBluetoothSignal,
+  createBleAdvertiser,
+  getBluetoothState,
+  stopBroadcastBluetoothSignal,
+} from '../services/bluetoothService';
+import {cancelTasks} from './helperSaga';
+import {Action} from '../actions';
+import {Alert} from 'react-native';
+import {selectNetworkConnectionState} from '../selectors/lasnSelectors';
+
+export const BROADCAST_FREQUENCY = 10000; // in ms, 10 secs
+export const UNLINK_TIMEOUT = 60000; // in ms, 60 secs
+function* bluetoothStateEventHandler(event) {
+  switch (event.type) {
+    case BLUETOOTH_EVENT_TYPES.DEVICE_BLUETOOTH_STATE_CHANGED: {
+      const bluetoothStateAction = new Action(
+        BLUETOOTH_STATE_CHANGED,
+        event.payload,
+      );
+      yield put({...bluetoothStateAction});
+      const isBluetoothPoweredOff =
+        event.payload === BLUETOOTH_STATE.PoweredOff;
+      if (isBluetoothPoweredOff) {
+        const disconnectAction = new Action(START_DISCONNECTION);
+        yield put({...disconnectAction});
+      }
+    }
+  }
+}
+
+export function* bluetoothStateWatcher(bleManager) {
+  yield call(getInitialBluetoothState, bleManager); // get initial state!
+  const bluetoothStateChannel = yield call(createBluetoothChannel, bleManager);
+  while (true) {
+    const event = yield take(bluetoothStateChannel);
+    yield call(bluetoothStateEventHandler, event);
+  }
+}
+
+export function* getInitialBluetoothState(bleManager) {
+  const bluetoothState = yield call(getBluetoothState, bleManager);
+  const bluetoothAction = new Action(
+    CHECK_DEVICE_BLUETOOTH_STATE,
+    bluetoothState,
+  );
+  yield put({...bluetoothAction});
```

Comment:
Isn't this equivalent?
```suggestion
  yield put(bluetoothAction);
```